py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py

```diff
@@ -300,6 +300,8 @@ def _populate_trt_builder_config(
         if tactic_sources is not None:
             builder_config.set_tactic_sources(tactic_sources=tactic_sources)
 
+        builder_config.set_flag(trt.BuilderFlag.PREFER_PRECISION_CONSTRAINTS)
```

Comment:
@peri044 Thoughts on this flag? How would it interact with your strict types work?

Reply:
This is weak typing in TensorRT and conflicts with strong typing. The way we should do this (via [strong typing](https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#strongly-typed-networks)) is as follows 
In the decomposition, 
```
if half_to_float:
    x=x.to(torch.float32) # Doing this explicitly so that the cast node is captured.

return torch.log(
        torch.softmax(x, dim)
    )
```
Along with this, we need to enable `STRONGLY_TYPED` in the builder. We also shouldn't use `layer.precision` in strong typing case. We should prefer strong typing since that is what TRT recommends.

Reply:
Support for strong typing is in this PR : https://github.com/pytorch/TensorRT/pull/3110

Reply:
@HolyWu The PR for strong typing is now merged. Could you take a look and make necessary changes for this PR to work ?

Reply:
I have reverted the `set_flag` and `layer.precision` change. I leave the node casting in `softmax` converter so that it works for both `softmax` and `log_softmax`. Please review.